Sources/SemanticIndex/SemanticIndexManager.swift

```diff
@@ -421,33 +397,60 @@ package final actor SemanticIndexManager {
   ///
   /// If `files` contains a header file, this will return a `FileToIndex` that re-indexes a main file which includes the
   /// header file to update the header file's index.
+  ///
+  /// The returned results will not include files that are known to be up-to-date based on `indexStoreUpToDateTracker`
+  /// or the unit timestamp will not be re-indexed unless `indexFilesWithUpToDateUnit` is `true`.
   private func filesToIndex(
-    toCover files: some Collection<DocumentURI> & Sendable
+    toCover files: some Collection<DocumentURI> & Sendable,
+    indexFilesWithUpToDateUnits: Bool
   ) async -> [FileToIndex] {
     let sourceFiles = await orLog("Getting source files in project") {
       Set(try await buildSystemManager.sourceFiles(includeNonBuildableFiles: false).keys)
```

Comment:
I know this was here before, but this could be fairly large depending on the project - probably not great to reconstruct the set every time this is called.

Reply:
Good idea. Caching the set now.